packages/undo-manager/src/types.ts

```diff
@@ -0,0 +1,19 @@
+export type HistoryChange = {
+	from: any;
+	to: any;
+};
+
+export type HistoryChanges = {
+	id: string | Record< string, any >;
```

Comment:
If the "composite" key was `Record< string, string | number >` instead of using `any`, would it work?

Reply:
For our use-case yes. the important thing is to be able to shallow compare the keys, so in that sense `any` is kind of appropriate. 

I get a sense that we feel `any` should be discouraged entirely. IMO, any as used in the types here is appropriate.

Reply:
Yes, you're probably right that `any` is fine. The types already tell us a lot about the API and the expected structure of objects (something must be an array, something has `from` and `to` fields, ...), and trying to type everything 100% often leads to incomprehensible generic monsters.